sparta/cache/Cache.hpp

```diff
@@ -174,6 +174,12 @@ namespace sparta
                 return peekCacheSet(addr).findInvalidWay();
             }
 
+            // Search for invalid in user-defined order of ways.
+            uint32_t findInvalidWay(uint64_t addr, std::vector<uint32_t> way_order) const
```

Comment:
Make parameter "const std::vector<uint32_t> way_order& way_order)" to avoid the copy